signer/Cargo.toml

```diff
@@ -26,34 +27,35 @@ ecdsa = ["secp256k1"]
 
 # Make the keypair algorithms here compatible with Subxt's Signer trait,
 # so that they can be used to sign transactions for compatible chains.
-subxt = ["dep:subxt"]
+subxt = ["dep:subxt-core"]
 
 # The getrandom package is used via schnorrkel. We need to enable the JS
 # feature on it if compiling for the web.
-web = ["getrandom/js", "subxt?/web"]
-native = ["subxt?/native"]
+web = ["getrandom/js"]
 
 [dependencies]
-subxt = { workspace = true, optional = true, default-features = false }
-regex = { workspace = true }
+subxt-core = { workspace = true, optional = true, default-features = false }
+regex = { workspace = true, default-features = false }
 hex = { workspace = true }
+cfg-if = { workspace = true }
 codec = { package = "parity-scale-codec", workspace = true, features = ["derive"] }
-sp-core-hashing = { workspace = true }
-thiserror = { workspace = true }
-pbkdf2 = { workspace = true }
-sha2 = { workspace = true }
-hmac = { workspace = true }
+sp-core-hashing = { workspace = true, default-features = false }
+derive_more = { workspace = true }
+pbkdf2 = { workspace = true, default-features = false }
+sha2 = { workspace = true, default-features = false }
+hmac = { workspace = true, default-features = false }
 zeroize = { workspace = true }
-bip39 = { workspace = true }
-schnorrkel = { workspace = true, optional = true }
-secp256k1 = { workspace = true, features = ["recovery", "global-context"], optional = true }
+bip39 = { workspace = true, default-features = false }
+schnorrkel = { workspace = true, optional = true, default-features = false }
+secp256k1 = { workspace = true, optional = true, default-features = false, features = ["alloc", "recovery"] } # features = ["recovery", "global-context"], 
```

Comment:
why is "global-context" removed?

Reply:
IIRC "global-context" is not compatible with `no-std` and we use `Secp256k1::signing_only()` and `Secp256k1::verification_only()` to get an appropriate context where needed.